src/ChannelServer/Database/Account.cs

```diff
@@ -90,6 +96,32 @@ public ChatMacro[] GetChatMacros()
 		}
 
 		/// <summary>
+		/// Adds a revealed map to the account.
+		/// </summary>
+		/// <param name="map"></param>
+		public void AddRevealedMap(RevealedMap revealedMap)
+		{
+			lock (_revealedMaps)
+			{
+				var oldMap = _revealedMaps.FirstOrDefault(x => x.MapId == revealedMap.MapId);
```

Comment:
I don't know how often this will be called, but a Dictionary and TryGetValue might be more efficient than repeatedly calling FirstOrDefault on a potentially large list.

Reply:
It's called once per login to the zone server. However, it's still a good idea to use an efficient implementation if one exists.

Reply:
Isn't it also called to update the data though while you run around? In that case it might be called much more often.

Reply:
Oh yes you are correct. I was looking at the `Get` method. The sending of maps is done only once.